include/cetl/rtti.hpp

```diff
@@ -0,0 +1,297 @@
+/// @file
+/// A simple runtime type information (RTTI) support designed for high-integrity
+/// real-time systems, where the use of the standard C++ RTTI is discouraged.
```

Comment:
How about this for the header comment:

> An explicit mechanism for implementing dynamic type introspection for high-integrity systems. This should be used instead of standard C++ RunTime Type Information (RTTI)  (i.e. `-f-nortti`) where a codebase must trace from lines of source back to verifications (correctness) and requirements (suitability).